vortex-serde/src/chunked_reader/mod.rs

```diff
@@ -52,8 +52,10 @@ impl<R: VortexReadAt> ChunkedArrayReader<R> {
         Ok(())
     }
 
+    // Making a new ArrayStream requires us to clone the reader to make
+    // multiple streams that can each use the reader.
     pub async fn array_stream(&mut self) -> impl ArrayStream + '_ {
-        let mut cursor = Cursor::new(&self.read);
+        let mut cursor = Cursor::new(self.read.clone());
```

Comment:
basically what the comment says: to be able to make sure the future is `'static`, I need to make sure that the reader is `'static` so we need to hand out owned-references (in this case `Arc`) rather than regular references